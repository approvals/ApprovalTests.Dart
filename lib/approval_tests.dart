import 'dart:convert';
import 'dart:io';
import 'dart:math';
import 'dart:mirrors';

import 'package:dcli/dcli.dart';
import 'package:talker/talker.dart';

part 'src/approvals.dart';
part 'src/writers/approval_text_writer.dart';
part 'src/core/logger/logger.dart';
part 'src/core/utils/utils.dart';
part 'src/core/options.dart';
part 'src/namer/namer.dart';
part 'src/comparator/comparator.dart';
part 'src/core/enums/comporator_ide.dart';
part 'src/comparator/command_line_comparator.dart';
part 'src/comparator/ide_comparator.dart';
part 'src/reporter/doesnt_match_exception.dart';
part 'src/core/utils/extensions.dart';
part 'src/core/utils/converter.dart';
part 'src/core/utils/executable_query.dart';
part 'src/namer/approval_number.dart';
