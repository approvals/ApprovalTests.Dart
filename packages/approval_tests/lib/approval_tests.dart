// ignore_for_file: implementation_imports

import 'dart:convert';
import 'dart:io';
import 'dart:math';

import 'package:diff_match_patch2/diff_match_patch.dart';
import 'package:talker/talker.dart';
import 'package:test_api/src/backend/invoker.dart' show Invoker;

part 'src/approvals.dart';
part 'src/comparator/file_comparator.dart';
part 'src/core/approval_number.dart';
part 'src/core/approval_writer.dart';
part 'src/core/comparator.dart';
part 'src/core/enums/comporator_ide.dart';
part 'src/core/extensions/approval_string_extensions.dart';
part 'src/core/utils/file_path_extractor.dart';
part 'src/core/logger/logger.dart';
part 'src/core/options.dart';
part 'src/core/platform_wrapper.dart';
part 'src/core/reporter.dart';
part 'src/core/scrubber.dart';
part 'src/core/stack_trace_fetcher/fetcher.dart';
part 'src/core/stack_trace_fetcher/stack_trace_fetcher.dart';
part 'src/core/utils/converter.dart';
part 'src/core/utils/executable_query.dart';
part 'src/core/utils/utils.dart';
part 'src/exceptions/doesnt_match_exception.dart';
part 'src/exceptions/file_not_found_exception.dart';
part 'src/exceptions/no_diff_tool_exception.dart';
part 'src/namer/file_namer_options.dart';
part 'src/namer/namer.dart';
part 'src/reporters/command_line/command_line_reporter.dart';
part 'src/reporters/diff_tool/diff_info.dart';
part 'src/reporters/diff_tool/diff_tool_reporter.dart';
part 'src/reporters/diff_tool/diff_tools.dart';
part 'src/scrubbers/date_scrubber.dart';
part 'src/scrubbers/nothing_scrubber.dart';
part 'src/scrubbers/reg_exp_scrubber.dart';
part 'src/writers/approval_text_writer.dart';
part 'src/reporters/git.dart';
