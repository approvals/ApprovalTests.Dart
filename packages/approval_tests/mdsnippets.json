{
  "$schema": "https://raw.githubusercontent.com/SimonCropp/MarkdownSnippets/master/schema.json",
  "Convention": "InPlaceOverwrite",
  "TocLevel": 5
}
