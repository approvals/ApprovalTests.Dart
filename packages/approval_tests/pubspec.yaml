#  * 1.0.0 for a first major release.
#  * 0.1.0 for a breaking release.
#  * 0.0.4 for a minor release.
name: approval_tests
description: Approval Tests implementation in Dart. Inspired by ApprovalTests.
version: 1.0.0
repository: https://github.com/approvals/ApprovalTests.Dart
homepage: https://github.com/approvals/ApprovalTests.Dart.git
issue_tracker: https://github.com/approvals/ApprovalTests.Dart/issues

topics:
  - test
  - approval
  - approvals
  - tests
  - approval-tests

environment:
    sdk: '>=3.0.0 <4.0.0'

dependencies:
  test: ^1.25.2
  talker: ^4.3.2
  diff_match_patch2: ^0.5.0
  test_api: ^0.7.0

dev_dependencies:
  sizzle_lints: ^2.0.2