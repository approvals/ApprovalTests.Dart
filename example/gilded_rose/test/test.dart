// ignore_for_file: avoid_relative_lib_imports

import 'package:approval_tests/approval_tests.dart';
import 'package:test/test.dart';

import '../lib/gilded_rose.dart';
import '../lib/item.dart';

void main() {
  // Define all test cases
  const allTestCases = [
    [
      "foo",
      "Aged Brie",
      "Backstage passes to a TAFKAL80ETC concert",
      "Sulfuras, Hand of Ragnaros",
    ],
    [-1, 0, 5, 6, 10, 11],
    [-1, 0, 1, 49, 50],
  ];

  group('Approval Tests for Gilded Rose', () {
    test('should verify all combinations of test cases', () {
      // Perform the verification for all combinations
      Approvals.verifyAllCombinations(
        allTestCases,
        options: const Options(
          comparator: IDEComparator(
            ide: ComparatorIDE.visualStudioCode,
          ),
          filesPath: "example/gilded_rose/test/approved_results/test",
        ),
        processor: processItemCombination,
      );
    });
  });
}

// Function to process each combination and generate output for verification
String processItemCombination(Iterable<List<dynamic>> combinations) {
  final receivedBuffer = StringBuffer();

  for (final combination in combinations) {
    // Extract data from the current combination
    final String itemName = combination[0] as String;
    final int sellIn = combination[1] as int;
    final int quality = combination[2] as int;

    // Create an Item object representing the current combination
    final Item testItem = Item(itemName, sellIn: sellIn, quality: quality);

    // Passing testItem to the application
    final GildedRose app = GildedRose(items: [testItem]);

    // Updating quality of testItem
    app.updateQuality();

    // Adding the updated item to expectedItems
    receivedBuffer.writeln(testItem.toString());
  }

  // Return a string representation of the updated item
  return receivedBuffer.toString();
}
