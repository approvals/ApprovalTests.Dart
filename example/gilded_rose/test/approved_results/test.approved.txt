Item{name: foo, sellIn: -2, quality: -1}
Item{name: foo, sellIn: -2, quality: 0}
Item{name: foo, sellIn: -2, quality: 0}
Item{name: foo, sellIn: -2, quality: 47}
Item{name: foo, sellIn: -2, quality: 48}
Item{name: foo, sellIn: -1, quality: -1}
Item{name: foo, sellIn: -1, quality: 0}
Item{name: foo, sellIn: -1, quality: 0}
Item{name: foo, sellIn: -1, quality: 47}
Item{name: foo, sellIn: -1, quality: 48}
Item{name: foo, sellIn: 4, quality: -1}
Item{name: foo, sellIn: 4, quality: 0}
Item{name: foo, sellIn: 4, quality: 0}
Item{name: foo, sellIn: 4, quality: 48}
Item{name: foo, sellIn: 4, quality: 49}
Item{name: foo, sellIn: 5, quality: -1}
Item{name: foo, sellIn: 5, quality: 0}
Item{name: foo, sellIn: 5, quality: 0}
Item{name: foo, sellIn: 5, quality: 48}
Item{name: foo, sellIn: 5, quality: 49}
Item{name: foo, sellIn: 9, quality: -1}
Item{name: foo, sellIn: 9, quality: 0}
Item{name: foo, sellIn: 9, quality: 0}
Item{name: foo, sellIn: 9, quality: 48}
Item{name: foo, sellIn: 9, quality: 49}
Item{name: foo, sellIn: 10, quality: -1}
Item{name: foo, sellIn: 10, quality: 0}
Item{name: foo, sellIn: 10, quality: 0}
Item{name: foo, sellIn: 10, quality: 48}
Item{name: foo, sellIn: 10, quality: 49}
Item{name: Aged Brie, sellIn: -2, quality: 1}
Item{name: Aged Brie, sellIn: -2, quality: 2}
Item{name: Aged Brie, sellIn: -2, quality: 3}
Item{name: Aged Brie, sellIn: -2, quality: 50}
Item{name: Aged Brie, sellIn: -2, quality: 50}
Item{name: Aged Brie, sellIn: -1, quality: 1}
Item{name: Aged Brie, sellIn: -1, quality: 2}
Item{name: Aged Brie, sellIn: -1, quality: 3}
Item{name: Aged Brie, sellIn: -1, quality: 50}
Item{name: Aged Brie, sellIn: -1, quality: 50}
Item{name: Aged Brie, sellIn: 4, quality: 0}
Item{name: Aged Brie, sellIn: 4, quality: 1}
Item{name: Aged Brie, sellIn: 4, quality: 2}
Item{name: Aged Brie, sellIn: 4, quality: 50}
Item{name: Aged Brie, sellIn: 4, quality: 50}
Item{name: Aged Brie, sellIn: 5, quality: 0}
Item{name: Aged Brie, sellIn: 5, quality: 1}
Item{name: Aged Brie, sellIn: 5, quality: 2}
Item{name: Aged Brie, sellIn: 5, quality: 50}
Item{name: Aged Brie, sellIn: 5, quality: 50}
Item{name: Aged Brie, sellIn: 9, quality: 0}
Item{name: Aged Brie, sellIn: 9, quality: 1}
Item{name: Aged Brie, sellIn: 9, quality: 2}
Item{name: Aged Brie, sellIn: 9, quality: 50}
Item{name: Aged Brie, sellIn: 9, quality: 50}
Item{name: Aged Brie, sellIn: 10, quality: 0}
Item{name: Aged Brie, sellIn: 10, quality: 1}
Item{name: Aged Brie, sellIn: 10, quality: 2}
Item{name: Aged Brie, sellIn: 10, quality: 50}
Item{name: Aged Brie, sellIn: 10, quality: 50}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: -2, quality: 0}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: -2, quality: 0}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: -2, quality: 0}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: -2, quality: 0}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: -2, quality: 0}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: -1, quality: 0}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: -1, quality: 0}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: -1, quality: 0}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: -1, quality: 0}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: -1, quality: 0}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 4, quality: 2}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 4, quality: 3}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 4, quality: 4}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 4, quality: 50}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 4, quality: 50}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 5, quality: 1}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 5, quality: 2}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 5, quality: 3}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 5, quality: 50}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 5, quality: 50}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 9, quality: 1}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 9, quality: 2}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 9, quality: 3}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 9, quality: 50}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 9, quality: 50}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 10, quality: 0}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 10, quality: 1}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 10, quality: 2}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 10, quality: 50}
Item{name: Backstage passes to a TAFKAL80ETC concert, sellIn: 10, quality: 50}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: -1, quality: -1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: -1, quality: 0}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: -1, quality: 1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: -1, quality: 49}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: -1, quality: 50}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 0, quality: -1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 0, quality: 0}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 0, quality: 1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 0, quality: 49}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 0, quality: 50}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 5, quality: -1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 5, quality: 0}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 5, quality: 1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 5, quality: 49}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 5, quality: 50}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 6, quality: -1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 6, quality: 0}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 6, quality: 1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 6, quality: 49}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 6, quality: 50}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 10, quality: -1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 10, quality: 0}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 10, quality: 1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 10, quality: 49}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 10, quality: 50}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 11, quality: -1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 11, quality: 0}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 11, quality: 1}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 11, quality: 49}
Item{name: Sulfuras, Hand of Ragnaros, sellIn: 11, quality: 50}
